library/src/updater.rs

```diff
@@ -89,26 +91,24 @@ pub fn init(app_config: AppConfig, yaml: &str) -> Result<(), UpdateError> {
     set_config(app_config, config).map_err(|err| UpdateError::InvalidState(err.to_string()))
 }
 
-fn check_for_update_internal(config: &ResolvedConfig) -> bool {
-    // Load UpdaterState from disk
-    // If there is no state, make an empty state.
-    let state = UpdaterState::load_or_new_on_error(&config.cache_dir, &config.release_version);
-    // Send info from app + current slot to server.
-    let response_result = send_patch_check_request(&config, &state);
-    match response_result {
-        Err(err) => {
-            error!("Failed update check: {err}");
-            return false;
-        }
-        Ok(response) => {
-            return response.patch_available;
-        }
-    }
-}
-
 /// Synchronously checks for an update and returns true if an update is available.
-pub fn check_for_update() -> bool {
-    return with_config(check_for_update_internal);
+pub fn check_for_update() -> anyhow::Result<bool> {
+    with_config(|config| {
+        // Load UpdaterState from disk
+        // If there is no state, make an empty state.
+        let state = UpdaterState::load_or_new_on_error(&config.cache_dir, &config.release_version);
+        // Send info from app + current slot to server.
+        let response_result = send_patch_check_request(&config, &state);
+        match response_result {
+            Err(err) => {
```

Comment:
In a follow up patch, we should just make the Rust API return these errors, and the c_api responsible to log them when needed.